### ISSUE TITLE
Rust: Use of undefined Actor value matches Any

### ISSUE THREAD

**Issue by dgrijalva** [osohq/oso#1573]
We're trying to use an enum for our Action values, to prevent reduce the opportunity for typo style mistakes. In setting this up, we were surprised to discover that the default behavior for an undefined Action value is to have it match any action. We actually encountered this error a few different ways:

* Forgetting to `register_class` the enum type causes all references to that type to match any action
* Any typo in the polar file will also match any action

These are both examples of user error, but likely to be pretty common. In my opinion, using an undefined constant for the action should either produce an error or match no actions. I'd be happy to work up a PR, if that's helpful.

Example: https://gist.github.com/dgrijalva/fd02d7cc746cb3b86db81aa0364c791e
Slack Thread: https://oso-oss.slack.com/archives/C017KSBTD4M/p1653673162892689

**Comment by samscott89** [comment]
Hey @dgrijalva !

Thanks for opening an issue. You're right, that's definitely not good behaviour.

The root cause of this, is that `Foo::Bar` is simply treated as an ident (name of a variable). And so `allow(auth: RequestAuth, AuthAction::TYPO, resource: Foo)` is treated like `allow(auth: RequestAuth, some_variable, resource: Foo)`.

And `some_variable` is not bound to anything, and so matches _any_ action.

Now, normally we catch this kind of mistake. E.g. if you write `allow(user, "read", doc) if usre.id = doc.owner_id` (note the `usre` typo), we'll raise this as an error -- we tell you that `user` is a singleton variable, it's only used in one place.

Except, we treat variable names with `::` in them specially... I think this is faulty logic that dates back to a time when we didn't always know if a variable was a constant/class or not.

Long story short, I think if we delete this line of code: https://github.com/osohq/oso/blob/2656313dc46581a588fcf0c57dee7bf91afbb3d0/polar-core/src/validations.rs#L52 then this will fix the issue.

If you'd be willing to submit a PR to fix this, we'd love it. I think you could extend the existing singleton variable test case that we have here: https://github.com/osohq/oso/blob/2656313dc46581a588fcf0c57dee7bf91afbb3d0/polar-core/tests/integration_tests.rs#L1621-L1624

**Comment by dgrijalva** [comment]
Thanks for all that info. I'll have a look and see if I can come up with a useful PR.
